### PR TITLE
feat!: implement versioned transactions

### LINKED ISSUE #694: Update Jupiter API
Update backend swap to use new Jupiter API v4

Checklist:
- [ ] Update Solana library to support Versioned Transactions
- [ ] Update Jupiter API to v4
- [ ] Update backend to use new api

### PR DESCRIPTION
**Pull request by justinenerio**
## Changes

- updated rpc client specs
- added support for Versioned Transactions and Address Lookup Tables
- updated deprecated rpc methods in cryptoplease 

## Related issues

Fixes #694

## Checklist

- [x] PR is ready for review (if not, it should be a draft).
- [x] PR title follows [Conventional Commits][1] guidelines.
- [ ] Screenshots/video added.
- [x] Tests added.
- [x] Self-review done.

[1]: https://www.conventionalcommits.org/en/v1.0.0/

### CODE REVIEW

File: packages/solana/lib/src/encoder/message/compiled_keys.dart [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+    required this.isWritable,
+    required this.isInvoked,
+  });
+
+  final bool isSigner;
+  final bool isWritable;
+  final bool isInvoked;
+
+  @override
+  bool operator ==(Object other) =>
```
Comment (ookami-kb) [inline packages/solana/lib/src/encoder/message/compiled_keys.dart:20]: Use freezed instead. Same for other similar classes.

**Review by ookami-kb — CHANGES_REQUESTED** [review]

File: packages/solana/lib/src/encoder/address_lookup_table/address_lookup_state.dart [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  const factory AddressLookupTableState({
+    required BigInt deactivationSlot,
+    required int lastExtendedSlot,
+    required int lastExtendedSlotStartIndex,
+    required Ed25519HDPublicKey? authority,
+    required List<Ed25519HDPublicKey> addresses,
+  }) = _AddressLookupTableState;
+}
+
+// class AddressLookupTableState {
```
Comment (ookami-kb) [inline packages/solana/lib/src/encoder/address_lookup_table/address_lookup_state.dart:14]: Please remove commented code.

**Review by ookami-kb — CHANGES_REQUESTED** [review]

File: packages/solana/lib/src/encoder/signed_tx.dart [hunk 3]
```diff
@@ -53,11 +66,27 @@ class SignedTx {
 
   String get blockhash => _txData.blockhash;
 
-  late final Message message = Message(instructions: _txData.instructions);
+  late final Message message =
+      Message(instructions: (_txData as TxLegacy).instructions);
```
Comment (ookami-kb) [inline packages/solana/lib/src/encoder/signed_tx.dart:70]: Why does it suppose that it's always `TxLegacy`?

File: packages/solana/lib/src/encoder/signed_tx.dart [hunk 4]
```diff
@@ -39,11 +38,25 @@ class SignedTx {
     );
 
     final messageBytes = reader.buf.buffer.asUint8List(reader.offset);
-    final txData = _TxData.decompile(messageBytes);
 
-    final signatures = signaturesData.mapIndexed(
-      (i, s) => Signature(s, publicKey: txData.accounts[i].pubKey),
-    );
+    final prefix = messageBytes.first;
+    final maskedPrefix = prefix & 0x7f;
```
Comment (ookami-kb) [inline packages/solana/lib/src/encoder/signed_tx.dart:43]: Make `0x7f` a file-private constant, it's used more that in one place.

**Review by ookami-kb — CHANGES_REQUESTED** [review]

File: packages/solana/lib/src/encoder/message/decode_message_args.dart [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:solana/src/encoder/address_lookup_table/address_lookup_table.dart';
+import 'package:solana/src/encoder/message/loaded_addresses.dart';
+
+part 'decode_message_args.freezed.dart';
+
+@freezed
+class DecodeMessageArgs with _$DecodeMessageArgs {
+  const factory DecodeMessageArgs.accountKeys(
```
Comment (ookami-kb) [inline packages/solana/lib/src/encoder/message/decode_message_args.dart:9]: Where does it come from at all? I don't see any usages of it. What is the use case?

Comment (justinenerio) [reply packages/solana/lib/src/encoder/message/decode_message_args.dart:9]: This union class is used to decode the transaction, it is used here: https://github.com/espresso-cash/espresso-cash-public/blob/241d8930ffb5dceb89bc2bd57a2f156b7094ca5b/packages/solana/lib/src/encoder/signed_tx.dart#L69

**Review by ookami-kb — APPROVED** [review]

**Pull request merged**